client/store/reports/utils.js

```diff
@@ -110,6 +110,24 @@ export function isReportDataEmpty( report ) {
 	return false;
 }
 
+/**
+ * Returns true if `date` is today.
+ *
+ * @param  {String|Object} date Date string or object in a format that can be
+ * understood by the Date JS object. For example `yyyy-mm-dd`.
+ * @returns {Boolean} Whether `date` is today.
+ */
+function isToday( date ) {
+	const d = new Date( date );
+	const today = new Date();
+
+	return (
+		d.getFullYear() === today.getFullYear() &&
+		d.getMonth() === today.getMonth() &&
+		d.getDate() === today.getDate()
+	);
+}
+
```

Comment:
Should this go in `/packages/date`? If so, it would need some testing.

Alternatively, below in line 143 you could keep it all in moment.js land. `getCurrentDates` always returns a Moment object for before and after:

https://github.com/woocommerce/wc-admin/blob/8b6143b79f8715b7aa19446b05d0b11d988503db/packages/date/src/index.js#L23-L24

```js
const end = datesFromQuery[ dataType ].before;
const endingTimeOfDay = end.isSame( moment(), 'day' ) ? 'now' : 'end';
```

Reply:
> Alternatively, below in line 143 you could keep it all in moment.js land.

I like this solution! However, I noticed `getCurrentDates` was not returning Moment objects but string dates. I [fixed](https://github.com/woocommerce/wc-admin/pull/1229/commits/53a1640ef1ae2f936089e5cbaeaf19ff67f14c15) that and updated `appendTimestamp` to accept Moment objects so everything keeps working as usual.

I know you pre-aproved this, but since the changes were bigger than expected, you might want to take another look at this PR. :slightly_smiling_face: